scripts/managed-services/get-kafka-instance-state/get-kafka-instance-state.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -o nounset
+set -o pipefail
+
+function main() {
+  local managedKafkaCR
+  managedKafkaCR=$(oc -n "${resource_namespace}" get managedkafka -o name)
+  local kafkaCR
+  kafkaCR=$(oc -n "${resource_namespace}" get kafka -o name)
+  local kafkaStatefulSet
+  kafkaStatefulSet=$(oc -n "${resource_namespace}" get statefulset -l app.kubernetes.io/name=kafka -o name)
+  local timestamp
+  timestamp=$(date +%s)
+  local inspectDir
+  inspectDir="inspect.${resource_namespace}.${timestamp}"
+
+  mkdir "${inspectDir}"
+
+  if [[ ! $(oc get namespace "${resource_namespace}") ]]; then
```

Comment:
Move this defensive code up to before you first use ${resource_namespace}, i.e. line6.